arkouda/alignment.py

```diff
@@ -172,6 +175,63 @@ def in1dmulti(a, b, assume_unique=False, symmetric=False):
             return atruth
 
 
+def find(query, space):
+    """
+    Return indices of query items in a search list of items (-1 if not found).
+
+    Parameters
+    ----------
+    query : (sequence of) array-like
+        The items to search for. If multiple arrays, each "row" is an item.
+    space : (sequence of) array-like
+        The set of items in which to search. Must have same shape/dtype as query.
+
+    Returns
+    -------
+    indices : pdarray, int64
+        For each item in query, its index in space or -1 if not found.
+    """
+
+    # Concatenate the space and query in fast (block interleaved) mode
+    if isinstance(query, (pdarray, Strings, Categorical)):
+        if type(query) != type(space):
+            raise TypeError("Arguments must have same type")
+        c = concatenate((space, query), ordered=False)
+        spacesize = space.size
+        querysize = query.size
+    else:
+        atypes = np.array([ai.dtype for ai in query])
+        btypes = np.array([bi.dtype for bi in space])
+        if not (atypes == btypes).all():
+            raise TypeError("Array dtypes of arguments must match")
+        c = [concatenate((si, qi), ordered=False) for si, qi in zip(space, query)]
+        spacesize = space[0].size
+        querysize = query[0].size
```

Comment:
is every element of `space`/`query` guaranteed to have the same size as the first?

Reply:
Another good point. Aside: it might be cool to create a `MultiArray` class that ensures all arrays are the same size and carries a `.size` attribute. Like a `DataFrame` without all the methods -- useful for all these functions that accept either arrays or lists of same-length arrays.